collections/_tutorials/2022-09-12-tutorial-cvae.md

```diff
@@ -0,0 +1,296 @@
+
+$$
+p(x,y,z) = \left\{
+  \begin{array}
+    pp(y,z \vert x) \cdot p(x) \\
+    p(y \vert x,z) \cdot p(x,z)
+  \end{array}
+  \right.
+$$
+
+$$p(x,y) = p(z \vert x) \cdot p(x)$$
+
+the next equations can be easily obtained
+
+$$p(y,z \vert x) = \frac{p(y \vert x,z) \cdot p(x,z)}{p(x)}$$
+
+$$p(y,z \vert x) = \frac{p(y \vert x,z) \cdot p(z \vert x) \cdot p(x)}{p(x)}$$
+
+$$p(y,z \vert x) = p(y \vert x,z) \cdot p(z \vert x)$$
+
+&nbsp;
+
+The ELBO $$\mathcal{L}$$ expression can thus be rewritten as
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot log\left(\frac{p(y \vert x,z) \cdot p(z \vert x)}{q(z \vert x,y)}\right) \,dz}$$
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot \left[log\left(p(y \vert x,z)\right) + log\left(\frac{p(z \vert x)}{q(z \vert x,y)}\right) \right] \,dz}$$
+
+$$\mathcal{L} = \int{q(z \vert x,y) \cdot log\left(p(y \vert x,z)\right) \,dz} \,+\, \int{q(z \vert x,y) \cdot log\left(\frac{p(z \vert x)}{q(z \vert x,y)}\right) \,dz}$$
+
+$$\mathcal{L} =  \mathbb{E}_{z\sim q(z \vert x,y)} \left[log\left(p(y \vert x,z)\right)\right] - D_{KL}\left(q(z \vert x,y)\parallel p(z \vert x)\right)$$
+
+where $$\mathbb{E}_{z\sim q(z \vert x,y)}$$ is the mathematical expectation with respect to $$q(z \vert x,y)$$. 
+
+&nbsp;
+
+At this stage of analysis, it is necessary to model $$p(y \vert x,y)$$ via an ***a priori distribution***. Let's first note that $$p(y \vert x,z)$$ is modeled by a neural network $$f(\cdot)$$ so that $$\hat{y}=f(x,z)$$. Since this function is deterministic, it will allow to model $$p\left(y \vert \hat{y}\right)$$. By approximating $$p\left(y \vert \hat{y}\right)$$ by a Bernoulli distribution, we have
+
+$$\mathbb{E}_{z\sim q(z \vert x,y)} \left[log\left(p(y \vert \hat{y})\right)\right] = \mathbb{E}_{z\sim q(z \vert x,y)} \left[log\left( {\hat{y}}^y \cdot \left(1-\hat{y}\right)^{1-y} \right)\right]$$
+
+$$ = \mathbb{E}_{z\sim q(z \vert x,y)} \left[ y \, log\left( \hat{y} \right) + \left(1-y\right) \, log\left(1-\hat{y}\right) \right]$$
+
+$$ = \mathbb{E}_{z\sim q(z \vert x,y)} \left[-CE\left( y,f\left(x,z\right)\right)\right]$$
+
+where $$CE(\cdot)$$ corresponds to the conventional ***cross entropy function*** !
+
+&nbsp;
+
+Thanks to this modeling, we are finally looking for:
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\max} \,\,\, \left( \mathbb{E}_{z\sim q(z \vert x,y)} [-CE\left( y,f\left(x,z\right)\right)] - D_{KL}(\underbrace{q(z \vert x,y)}_{g,h}\parallel \underbrace{p(z \vert x)}_{k,l}) \right)$$
+
+$$\left(f^*,g^*,h^*,k^*,l^*\right) = \underset{(f,g,h,k,l)}{\arg\min} \,\,\, \left( \mathbb{E}_{z\sim q(z \vert x,y)} [CE\left( y,f\left(x,z\right)\right)] + D_{KL}(\underbrace{q(z \vert x,y)}_{g,h}\parallel \underbrace{p(z \vert x)}_{k,l}) \right)$$
+
+&nbsp;
+
+## Various scenarios
+
+There are different ways to leverage the cVAE formalism depending on the prior modeling $$p(z \vert x)$$ and the content of the $$y$$ references.
+
+### Prior modeling
```

Comment:
juste avant on disait que le prior était p(y|x,z).  maintenant c'est p(z|x)?

Reply:
et c'est quand même bizzare d'appeler ça un a priori alors que la même distribution chez les VAE s'appelle un a posteriori.

Reply:
We discussed this point this morning. I also tried to avoid ambiguities by rewording different parts of the text.